src/main/java/app/AppBuilder.java

```diff
@@ -51,8 +53,7 @@ public class AppBuilder {
     private final ViewManagerModel viewManagerModel = new ViewManagerModel();
     private final ViewManager viewManager = new ViewManager(cardPanel, cardLayout, viewManagerModel);
 
-        private final InMemoryUserDataAccessObject userDataAccessObject = new InMemoryUserDataAccessObject();
-
+    private final InMemoryUserDataAccessObject inMemoryUserDataAccessObject = new InMemoryUserDataAccessObject();
```

Comment:
You might want to delete InMemoryUserDAO completely and implement all the interfaces in MongoLoginUserDAO since we are using mongo and not in memory to store user data